src/main/java/fr/insalyon/creatis/gasw/parser/GaswParser.java

```diff
@@ -489,4 +495,56 @@ private static void addName(
             content.append(separator).append(f.getName());
         }
     }
-}
+
+    public GaswInput getGaswInput(String applicationName, Map<String, String> inputsMap, String executableName,HashMap<Integer, String> inputid , HashMap<Integer, String> outputid, String invocationString, 
+    Map<String, String> resultDirectory, String jobId, List<URI> downloads)
+            throws URISyntaxException, FileNotFoundException, IOException, GaswException, SAXException {
+
+        List<String> parameters = new ArrayList<>();
+        List<GaswUpload> uploads = new ArrayList<>();
+        this.downloads = downloads;
+
+        for (GaswArgument argument : arguments) {
+            StringBuilder param = new StringBuilder();
+            if (argument.getOption() != null) {
+                param.append(argument.getOption());
+                param.append(" ");
+            }
+            if (argument.getHookup() == GaswArgument.Hookup.Input) {
+                String value = inputsMap.get(argument.getName());
+                if (argument.getType() == GaswArgument.Type.URI) {
+                    // If the value already is a URI, use it as is.  If not, it
+                    // is a lfn and the prefix is added.
+                    URI valueURI = new URI(
+                        GaswUtil.isUri(value) ? value : LFN_PREFIX + value);
+                    param.append(new File(valueURI.getPath()).getName());
+                } else {
+                    // Need to escape special characters to avoid bash errors.
+                    param.append(escapeSpecialBashCharacters(value));
+                }
+            } else {
+                GaswOutputArg output = (GaswOutputArg) argument;
+                String value = parseOutputTemplate(
+                    output.getTemplateParts(), inputsMap);
+                // If the value already is a URI, use it as is.  If not, it is a
+                // lfn and the prefix is added.
+                URI valueURI = new URI(
+                    GaswUtil.isUri(value) ? value : LFN_PREFIX + value);
+                uploads.add(new GaswUpload(valueURI, output.getReplicas()));
+                param.append(new File(valueURI.getPath()).getName());
+            }
+
+            if (!argument.isImplicit()) {
+                parameters.add(param.toString());
+            }           
+        }
+        // Check if "results-directory" exists in inputsMap and add to uploads
+        if (inputsMap.containsKey("results-directory")) {
+        String resultsDirectory = inputsMap.get("results-directory");
+        URI resultsURI = new URI(GaswUtil.isUri(resultsDirectory) ? resultsDirectory : LFN_PREFIX + resultsDirectory);
+        uploads.add(new GaswUpload(resultsURI, GaswConstants.numberOfReplicas));
+    }
+        return new GaswInput(applicationName, executableName, parameters, downloads, uploads,
```

Comment:
Is `applicationname` needed ?
what is in `parameters` ?
I don't think `gaswVariables` and `envVariables`. Are they not always empty ?
And i think gasw get the jobId from the currentdirectory name in GaswConfiguration, so i don't think it is necessary to give it also in gaswInput.